content/docs/guides/query-log.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Query Log
+sort_rank: 1
+---
+
+# Using the Prometheus query log
+
+Prometheus has the ability to log all the queries run by the engine to a log
+file. This guide demonstrates how to use that log file, which fields it does
+contain and provides advanced tips about how to operate the log file.
+
+The query log has been introduced in v2.16.0 and is disabled by default.
+
+## Enable the query log
+
+The query log can be enabled and disabled at runtime. It can therefore be
+activated when you want to investigate slownesses or high load on your
+Prometheus instance.
+
+To enable or disable the query log, two steps are needed:
+
+1. Adapt the configuration to add or remove the query log configuration.
+1. Reload the Prometheus server configuration.
+
+### Logging all the queries to a file
+
+This example demonstrates how to log all the queries to
+a file called `/var/log/prometheus/query.log`.
+
+First, let's ensure that Prometheus has write access to a directory
+`/var/log/prometheus`. We will assume that Prometheus is running under a user
+called Prometheus:
+
+```shell
+# mkdir /var/log/prometheus
+# chown prometheus: /var/log/prometheus
+```
+
+Then, adapt the `prometheus.yml` configuration file:
+
+```yaml
+global:
+  scrape_interval:     15s
+  evaluation_interval: 15s
+  query_log_file: /var/log/prometheus/query.log
+scrape_configs:
+  - job_name: 'prometheus'
+    - targets: ['localhost:9090']
+```
+
+And, reload the Prometheus configuration:
+
+```shell
+$ curl -X POST http://127.0.0.1:9090/-/reload
+```
+
+Or, if Prometheus is not launched with `--web.enable-lifecycle`:
+
+```shell
+$ killall -HUP prometheus
```

Comment:
You're making an assumption of OS here, the purpose of killall varies quite a bit.